pkg/server/options/authorization.go

```diff
@@ -79,11 +112,48 @@ func (s *Authorization) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&s.AlwaysAllowPaths, "authorization-always-allow-paths", s.AlwaysAllowPaths,
 		"A list of HTTP paths to skip during authorization, i.e. these are authorized without "+
 			"contacting the 'core' kubernetes server.")
+
+	// Only surface selected, webhook-related CLI flags
+
+	fs.StringVar(&s.Webhook.WebhookConfigFile, "authorization-webhook-config-file", s.Webhook.WebhookConfigFile,
+		"File with optional webhook configuration in kubeconfig format. The API server will query the remote service to determine access on the API server's secure port.")
+
+	fs.StringVar(&s.Webhook.WebhookVersion, "authorization-webhook-version", s.Webhook.WebhookVersion,
+		"The API version of the authorization.k8s.io SubjectAccessReview to send to and expect from the webhook.")
+
+	fs.DurationVar(&s.Webhook.WebhookCacheAuthorizedTTL, "authorization-webhook-cache-authorized-ttl", s.Webhook.WebhookCacheAuthorizedTTL,
+		"The duration to cache 'authorized' responses from the webhook authorizer.")
+
+	fs.DurationVar(&s.Webhook.WebhookCacheUnauthorizedTTL, "authorization-webhook-cache-unauthorized-ttl", s.Webhook.WebhookCacheUnauthorizedTTL,
+		"The duration to cache 'unauthorized' responses from the webhook authorizer.")
 }
 
-func (s *Authorization) ApplyTo(config *genericapiserver.Config, kubeInformers, globalKubeInformers kcpkubernetesinformers.SharedInformerFactory, kcpInformers, globalKcpInformers kcpinformers.SharedInformerFactory) error {
+func (s *Authorization) ApplyTo(ctx context.Context, config *genericapiserver.Config, kubeInformers, globalKubeInformers kcpkubernetesinformers.SharedInformerFactory, kcpInformers, globalKcpInformers kcpinformers.SharedInformerFactory) error {
 	var authorizers []authorizer.Authorizer
 
+	// re-use the authorizer from the generic control plane (this is only set for webhooks)
+	if webhook := s.Webhook; webhook != nil && webhook.WebhookConfigFile != "" {
+		authorizationConfig, err := webhook.ToAuthorizationConfig(informerfactoryhack.Wrap(kubeInformers))
+		if err != nil {
+			return err
+		}
+
+		if config.EgressSelector != nil {
+			egressDialer, err := config.EgressSelector.Lookup(egressselector.ControlPlane.AsNetworkContext())
+			if err != nil {
+				return err
+			}
+			authorizationConfig.CustomDial = egressDialer
+		}
+
+		authorizer, _, err := authorizationConfig.New(ctx, config.APIServerID)
```

Comment:
Is this `ctx` just initializing one or used for runtime? If the first one, can we cap it with ctxWithTimeout? As it might block all startup hanging without a clear error? If second - I don't think we can do anything :/

Reply:
The context given here is ultimately used to start a goroutine that does

```go
filesystem.WatchUntil(
		ctx,
		r.reloadInterval,
		r.initialConfig.ReloadFile,
		func() {
			r.checkFile(ctx)
		},
		func(err error) {
			klog.ErrorS(err, "watching authorization config file")
		},
	)
```

in the auto-reloading webhook authorizer (it reloads the given webhook-config-file automatically at runtime).